pvlib/modelchain.py

```diff
@@ -659,29 +684,45 @@ def no_spectral_loss(self):
         return self
 
     @property
-    def temp_model(self):
-        return self._temp_model
+    def temperature_model(self):
+        return self._temperature_model
 
-    @temp_model.setter
-    def temp_model(self, model):
+    @temperature_model.setter
+    def temperature_model(self, model):
         if model is None:
-            self._temp_model = self.infer_temp_model()
+            self._temperature_model = self.infer_temp_model()
         elif isinstance(model, str):
             model = model.lower()
             if model == 'sapm':
-                self._temp_model = self.sapm_temp
+                self._temperature_model = self.sapm_temp
+            elif model == 'pvsyst':
+                self._temperature_model = self.pvsyst_temp
             else:
-                raise ValueError(model + ' is not a valid temp model')
+                raise ValueError(model + ' is not a valid temperature model')
         else:
-            self._temp_model = partial(model, self)
+            self._temperature_model = partial(model, self)
 
     def infer_temp_model(self):
-        raise NotImplementedError
+        params = set(self.system.temperature_model_parameters.keys())
+        if set(['a', 'b', 'deltaT']) <= params:
+            return self.sapm_temp
+        elif set(['u_c', 'u_v']) <= params:
+            return self.pvsyst_temp
+        else:
+            raise ValueError('could not infer temperature model from '
+                             'system.temperature_module_parameters {}.'
+                             .format(self.system.temperature_model_parameters))
 
     def sapm_temp(self):
-        self.temps = self.system.sapm_celltemp(self.total_irrad['poa_global'],
-                                               self.weather['wind_speed'],
-                                               self.weather['temp_air'])
+        self.cell_temperature = self.system.sapm_celltemp(
```

Comment:
Change in attribute name also deserves a note in whats new. We could deprecate the old name but I don't think it's worth it. I ran into this in the `forecast.rst` documentation.

Reply:
Also update the "Assigns attributes" section of the `ModelChain.run_model` docstring